pilot/cmd/pilot-agent/main.go

```diff
@@ -450,6 +473,25 @@ func getDNSDomain(domain string) string {
 	return domain
 }
 
+// check if SDS UDS path and token path exist, if both exist, requests key/cert
+// using SDS instead of secret mount.
+func getSDSData() (bool, string) {
+	if !sdsEnabledVar.Get() {
+		return false, ""
+	}
+
+	// If sdsenabled env var is set but uds doesn't exist, treat sds as disabled.
+	if !waitForFile(sdsUDSPath, sdsUdsWaitTimeout) {
+		return false, ""
+	}
+
+	if _, err := os.Stat(trustworthyJWTPath); err == nil {
+		return true, trustworthyJWTPath
+	}
+
+	return true, jwtPath
```

Comment:
Stat that file too - you may be on a VM, where this is not present. If JWT not found - return "".

Reply:
(because we may be using SPIRE or other agent that doesn't rely on tokens )